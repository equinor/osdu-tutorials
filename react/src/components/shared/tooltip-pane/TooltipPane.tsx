import React, { ReactNode } from 'react';
import './styles.css';

interface Props {
  /** a tooltip content */
  children: ReactNode;

  /** apply common set of styles for a one-two lines of text */
  doApplyStylesForText?: boolean;

  /** an additional class name */
  className?: string;
}

export function TooltipPane({ children, doApplyStylesForText = false, className }: Props) {
  const classNameList = ['tooltip-pane'];
  if (doApplyStylesForText) {
    classNameList.push('tooltip-pane--for-text');
  }
  if (className) {
    classNameList.push(className);
  }
  return <div className={classNameList.join(' ')}>{children}</div>;
}
