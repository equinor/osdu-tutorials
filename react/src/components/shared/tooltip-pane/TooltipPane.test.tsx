import React from 'react';
import { shallow } from 'enzyme';
import { TooltipPane } from './TooltipPane';

describe('Loader', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<TooltipPane>text</TooltipPane>);

    expect(wrapper).toMatchSnapshot();
  });
});
