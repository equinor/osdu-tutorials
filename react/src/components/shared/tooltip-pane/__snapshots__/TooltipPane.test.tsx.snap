// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render self and subcomponents 1`] = `
<div
  className="tooltip-pane"
>
  text
</div>
`;
